### PR TITLE
Form questions

### PR DESCRIPTION
**Pull request by walaamedhat**
- Finished form questions using React and Redux

-  Editing on oAuth to send some data to the front-end 

- Redirect user if he newbie to the form page, else redirect to the profile page  

- Checked from cookie

### CODE REVIEW

**Comment by walaamedhat** [comment]
I apologize because the PR is very large  :neutral_face:

**Comment by MahmoudMH** [comment]
We need a column for name and another for username, for messages on gitter also for Github profile.
and the name naming The student on his profile

**Comment by walaamedhat** [comment]
We will store the username of GitHub instead of the name

File: src/client/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {
+  START,
+  SUCCESS
```
[the remark this answers is not in the document]
Comment (walaamedhat) [reply src/client/actions/index.js]: OK

File: src/server/index.js [hunk 2]
```diff
@@ -3,21 +3,82 @@ const path = require('path');
...
+app.use(passport.session());
+
+passport.use(new Strategy({
+
+  clientID: process.env.GITHUB_CLIENTID,
+  clientSecret: process.env.GITHUB_SECRET,
+  callbackURL: 'http://localhost:3000/auth/github/callback',
+  profileFields: ['email','displayName','profileUrl','picture.type(large)']
+},(accessToken,refreshToken,profile,done) => {
+  console.log(profile.username);
```
Comment (walaamedhat) [inline src/server/index.js]: remove console  :face_with_head_bandage:

File: .babelrc [hunk 3]
```diff
@@ -1,4 +1,4 @@
 {
-  "presets": [ "env", "react", "stage-1" ],
+  "presets": [ "env", "react"],
```
Comment (astroash) [inline .babelrc]: Why has this been removed? This is related to [public class field syntax](https://reactjs.org/docs/handling-events.html) as mentioned in gitter.

File: .travis.yml [hunk 4]
```diff
@@ -3,3 +3,7 @@ node_js:
 - "stable"
 script:
 - node tests.js
+
+notifications:
+  slack:
```
Comment (astroash) [inline .travis.yml]: Do you have slack notifications set up for travis? Also are you using travis? I thought you were using circleci?

File: src/client/components/navbar/navbar.css [hunk 5]
```diff
@@ -1,4 +1,4 @@
-body {
+/* body {
```
Comment (astroash) [inline src/client/components/navbar/navbar.css]: Why is this commented out? Can it be removed?

File: src/client/components/questions/q3.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    } else {
+      checkboxes.style.display = 'none';
+    }
+    this.setState({ showReply: !this.state.showReply });
+  }
+  getSkills() {
+    const skills =[];
+    const checkboxes = document.querySelectorAll('input[type=checkbox]:checked');
+
+    for (let i = 0; i < checkboxes.length; i++) {
```
Comment (astroash) [inline src/client/components/questions/q3.js]: Does this work? `skills` is a const and you are trying to mutate it. I recommend using a `.map` instead. Let me know if you want help with this.

File: src/client/index.css [hunk 7]
```diff
@@ -1,9 +1,15 @@
-*, *:before, *:after {
+/* *, *:before, *:after {
```
Comment (astroash) [inline src/client/index.css]: Why is this code commented out?

Comment (walaamedhat) [reply src/client/index.css]: Done :+1:

File: src/client/components/questions/q3.js (hunk 6 above)
Comment (walaamedhat) [reply src/client/components/questions/q3.js]: :smile:

File: src/client/components/navbar/navbar.css (hunk 5 above)
Comment (walaamedhat) [reply src/client/components/navbar/navbar.css]: OK

File: .travis.yml (hunk 4 above)
Comment (walaamedhat) [reply .travis.yml]: I will remove it

File: .babelrc (hunk 3 above)
Comment (walaamedhat) [reply .babelrc]: :smile:

**Pull request merged**